core/src/main/scala/kafka/tools/DumpLogSegments.scala

```diff
@@ -38,6 +38,7 @@ object DumpLogSegments {
     val parser = new OptionParser
     val printOpt = parser.accepts("print-data-log", "if set, printing the messages content when dumping data logs")
     val verifyOpt = parser.accepts("verify-index-only", "if set, just verify the index log without printing its content")
+    val indexSanityOpt = parser.accepts("verify-index-sanity", "if set, just verify the index sanity without printing its content")
```

Comment:
The sanity only checks 2 things

- The file size is a multiple of 8 as it only stores the offset num and the data file offset.
- The last offset < base offset in index file's name.

verify index only actually reads each index entry and tries to read the corresponding message from the data/log file.

In one of our customer environment we noticed that the --verify-index-only check was passing but the index-sanity check was failing which is kind of bizarre but they had file system related issues.  

When the index sanity check fails , the broker tries to automatically rebuild the index which again requires reading corresponding log file and is relatively expensive operation. In their case , they had 30 or 60 day retention and with lot of corrupt index files the broker restarts would take a long time. 

The sanity check is a cheap check so someone trying to figure out how many index files will be rebuilt if the broker is restarted can rather inexpensively find that out with this new option. I also plan to follow up with another jira where the index rebuilding can also be done via this tool rather than relying on broker restarts.